workflows/variantcalling.nf

```diff
@@ -11,12 +11,24 @@ WorkflowVariantcalling.initialise(params, log)
 
 // TODO nf-core: Add all file path parameters for the pipeline to the list below
 // Check input path parameters to see if they exist
-def checkPathParamList = [ params.input, params.fasta ]
+def checkPathParamList = [ params.input, params.fasta, params.fai, params.gzi, params.interval ]
 for (param in checkPathParamList) { if (param) { file(param, checkIfExists: true) } }
 
 // Check mandatory parameters
-if (params.input) { ch_input = file(params.input) } else { exit 1, 'Input samplesheet not specified!' }
+if (params.input)    { ch_input    = file(params.input)    } else { exit 1, 'Input samplesheet not specified!' }
 
+if (params.fasta)    { ch_fasta    = file(params.fasta)    } else { exit 1, 'Reference fasta not specified!' }
+if (params.fai)      { ch_fai      = file(params.fai)      } else { exit 1, 'Reference fasta index not specified!' }
+
+if (params.gzi) {
+    ch_gzi = file(params.gzi)
+} else if ( params.fasta.endsWith('fasta.gz') ) { 
+    exit 1, 'Reference fasta index gzi file not specified for fasta.gz file!' 
+}else {
+    ch_gzi = null
```

Comment:
Or maybe set it here, instead of `null` set it to `[]`.

Reply:
If it is a path, remove the `ch_` prefix from them. I agree, in that case in does not make sense. Why not create channels right there itself?

Reply:
remove all `ch_`, change to `_file` now.